tests/unittests/Foundation/WindowsOnly/NSPointerArrayInternalTests.mm

```diff
@@ -0,0 +1,114 @@
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <Foundation/Foundation.h>
+#import <TestFramework.h>
+#import <windows.h>
+
+// Expose some private functions of the NSPointerArray implementation
+@interface NSPointerArray (Private)
+- (NSUInteger)allocatedCount;
+- (const void* const)containerPtr;
+@end
+
+// Helper class, used to track deallocations
+@interface DummyTestClass : NSObject {
+    NSUInteger* _notify;
+}
+@end
+
+@implementation DummyTestClass
+- (instancetype)initWithNotifyPtr:(NSUInteger*)notify {
+    if (self = [super init]) {
+        _notify = notify;
+    }
+    return self;
+}
+
+- (void)dealloc {
+    *_notify = *_notify + 1;
+    [super dealloc];
+}
+@end
+
+TEST(NSPointerArray, Equality) {
+    NSPointerArray* arr1 = [NSPointerArray strongObjectsPointerArray];
+    [arr1 addPointer:@1];
+    [arr1 addPointer:nullptr];
+    [arr1 addPointer:@2];
+
+    NSPointerArray* arr2 = [NSPointerArray strongObjectsPointerArray];
+    [arr2 addPointer:@1];
+    [arr2 addPointer:nullptr];
+    [arr2 addPointer:@2];
+    BOOL equalArray1Array2 = [arr1 isEqual:arr2];
+    ASSERT_TRUE_MSG(equalArray1Array2, @"Both arrays should be equal");
+
+    NSPointerArray* arr3 = [NSPointerArray strongObjectsPointerArray];
+    [arr3 addPointer:@1];
+    [arr3 addPointer:@2];
+    [arr3 addPointer:nullptr];
+    BOOL equalArray1Array3 = [arr1 isEqual:arr3];
+    ASSERT_FALSE_MSG(equalArray1Array3, @"Both arrays should not be equal");
+
+    // Check the behaviour of the equality on C strings
+    const char* s1 = "first string";
+    const char* s2 = "second string";
+    const char* s3 = "third string";
+
+    NSPointerArray* cPtrArray1 = [NSPointerArray pointerArrayWithOptions:NSPointerFunctionsCStringPersonality];
+    [cPtrArray1 addPointer:nullptr];
+    [cPtrArray1 addPointer:const_cast<char*>(s1)];
+    [cPtrArray1 addPointer:const_cast<char*>(s2)];
+
+    char* dup_s1 = _strdup(s1);
+
+    NSPointerArray* cPtrArray2 = [NSPointerArray pointerArrayWithOptions:NSPointerFunctionsCStringPersonality];
+    [cPtrArray2 addPointer:nullptr];
+    [cPtrArray2 addPointer:dup_s1];
+    [cPtrArray2 addPointer:const_cast<char*>(s2)];
+    BOOL equalCStrA1A2 = [cPtrArray1 isEqual:cPtrArray2];
+
+    NSPointerArray* cPtrArray3 = [NSPointerArray pointerArrayWithOptions:NSPointerFunctionsCStringPersonality];
+    [cPtrArray3 addPointer:nullptr];
+    [cPtrArray3 addPointer:dup_s1];
+    [cPtrArray3 addPointer:const_cast<char*>(s3)];
+    BOOL equalCStrA1A3 = [cPtrArray1 isEqual:cPtrArray3];
```

Comment:
Was the intention to add the removed lines from NSPointerArrayTests.mm::109 here? Because this is still identical to what you have there, including missing those lines.